packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -962,6 +990,35 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
     }
   }
 
+  /// Collapses the selection at the given `offset` location.
+  ///
+  /// See also:
+  ///  * [_selectStartTo], which sets or updates selection start edge.
+  ///  * [_selectEndTo], which sets or updates selection end edge.
+  ///  * [_finalizeSelection], which stops the `continuous` updates.
+  ///  * [_clearSelection], which clear the ongoing selection.
+  ///  * [_selectWordAt], which selects a whole word at the location.
+  ///  * [selectAll], which selects the entire content.
+  void _collapseSelectionAtPosition({required Offset offset}) {
+    // Placing this before updating the start/end edge does not
+    // work to clear all unrelated selections because dispatching a selection edge
+    // update event will send the edge update to every selectable until arriving
+    // at the correct selectable, so the selection is set again after it is cleared.
+    // _clearSelection();
+
+    // _selectStartTo(offset: offset);
+    // _selectEndTo(offset: offset);
+
+    // This works to clear all unrelated selections but relies on a new selection event.
+    _selectable?.dispatchSelectionEvent(CollapseSelectionSelectionEvent(globalPosition: offset));
```

Comment:
Chun Heng should really have the final word, but I think dispatching a new event here seems fine...